cmd/snap/cmd_snap_op.go

```diff
@@ -357,20 +357,34 @@ func showDone(names []string, op string) error {
 		switch op {
 		case "install":
 			if snap.Developer != "" {
+				// TRANSLATORS: the args are a snap name optionally followed by a channel, then a version, then the developer name (e.g. "some-snap (beta) 1.3 from 'alice' installed")
 				fmt.Fprintf(Stdout, i18n.G("%s%s %s from '%s' installed\n"), snap.Name, channelStr, snap.Version, snap.Developer)
 			} else {
+				// TRANSLATORS: the args are a snap name optionally followed by a channel, then a version (e.g. "some-snap (beta) 1.3 installed")
 				fmt.Fprintf(Stdout, i18n.G("%s%s %s installed\n"), snap.Name, channelStr, snap.Version)
 			}
 		case "refresh":
 			if snap.Developer != "" {
+				// TRANSLATORS: the args are a snap name optionally followed by a channel, then a version, then the developer name (e.g. "some-snap (beta) 1.3 from 'alice' refreshed")
 				fmt.Fprintf(Stdout, i18n.G("%s%s %s from '%s' refreshed\n"), snap.Name, channelStr, snap.Version, snap.Developer)
 			} else {
+				// TRANSLATORS: the args are a snap name optionally followed by a channel, then a version (e.g. "some-snap (beta) 1.3 refreshed")
 				fmt.Fprintf(Stdout, i18n.G("%s%s %s refreshed\n"), snap.Name, channelStr, snap.Version)
 			}
+		case "revert":
+			// TRANSLATORS: first %s is a snap name, second %s is a revision
+			fmt.Fprintf(Stdout, i18n.G("%s reverted to %s\n"), snap.Name, snap.Version)
 		default:
 			fmt.Fprintf(Stdout, "internal error, unknown op %q", op)
 		}
+		if snap.TrackingChannel != snap.Channel {
+			// TRANSLATORS: first %s is a snap name, following two %s are channel names
+			fmt.Fprintf(Stdout, i18n.G("This leaves %s tracking %s, but the current revision is from %s.\n"),
```

Comment:
I have strong suspicions that this isn't going to work so well, and will end up confusing people. The problem is that snaps do not have an intrinsic channel. Something that is in candidate today can be in stable tomorrow, and we won't know the difference. The Channel field in SideInfo comes from a snapshot of when we observed the snap, but we don't know what happens next.

Reply:
OK, so how do we fix it? Users are getting confused by the behaviour _today_.

Reply:
(from the standup), i'll be dropping the mention of the channel "from", and the switch suggestion, the rest changes ok